### ISSUE TITLE
500 error: Quantile algorithm requested but no histogram querier is available. Make sure you have the appropriate data pipeline enabled.

### ISSUE THREAD

**Issue by mmurph3** [kubecost/cost-analyzer-helm-chart#2068]
Thanks for your interest in Kubecost!  To ensure you get the help you need, please follow these guidelines:

- If you have questions about product functionality/usage or need assistance with product installation/configuration, please email support@kubecost.com, where our support engineering team will be happy to help!
- If you have a reproducible bug or a feature request, you're in the right place.  Please include the information requested below.

---
# For reproducible bugs, please provide the following

**Describe the bug**  
Customer is on v1.101.1 and when they drill down into namespace and hit the redirect link to container right sizing OR when they go to container right sizing directly, they get this 500:
`Quantile algorithm requested but no histogram querier is available. Make sure you have the appropriate data pipeline enabled.`
I noticed they had commas in their custom request sizing recommendations profile under cpu/ram target utilizations.  I tried to repro but I can't, i have correctly formatted percentages. After some investigation, this is isolated to labels. Doesn't happen if namespace equals 'foo'

<img width="861" alt="image" src="https://user-images.githubusercontent.com/36980248/226365143-578c9e4a-82d5-480a-9ce9-876a90524ab9.png">

**To Reproduce**  
1. Go to to allocations
2. Click on ns
3. Click redirect link to request right sizing
-----
1. Savings > Container right sizing
2. Customer Request Sizing Recommendations
3. Add filters > labels equals <namespace>

**Expected behavior**  
Expect a redirect from allocations for ns to container right sizing for same ns with no 500 error and workloads to be displayed for ns.

**Screenshots**  
<img width="741" alt="image" src="https://user-images.githubusercontent.com/36980248/226365870-82a80f86-56db-4e6b-93bc-7bfc5f849da1.png">

<img width="353" alt="image" src="https://user-images.githubusercontent.com/36980248/226365967-619c5524-e162-43dc-98e5-0346295d9148.png">

**Comment by mmurph3** [comment]
They're also using EUR currency and get the commas in the percentages, I can't reproduce that locally.

`"currencyCode":"EUR"`

**Comment by kwombach12** [comment]
@Adam-Stack-PM can we add this to the 1.102 burndown?

**Comment by thomasvn** [comment]
Hey @mmurph3! I believe the quantile algorithm requires the [ContainerStats Pipeline](https://docs.kubecost.com/using-kubecost/getting-started/savings/auto-request-sizing/containerstats-pipeline) to be enabled in Kubecost.

Have you checked to see whether this is enabled?

**Comment by mmurph3** [comment]
@thomasvn waiting on customer. Thanks! I was also able to confirm this only happens when adding a label filter. Works fine with ns.

**Comment by mmurph3** [comment]
@thomasvn we no longer see this error but we do see the following when filtering by label:

`Recommending requests from data: CPU algorithm set to max but no max data is available`

opened an issue for it. I'm going to close this out and track the above error in the new issue I opened:
https://github.com/kubecost/cost-analyzer-helm-chart/issues/2104
